### DISCUSSION TITLE
Pages build and deployment queued for 10 hours and counting

### DISCUSSION THREAD

**Discussion by skibidibidop** [community/community#49074] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hi! I'd like to ask if anyone knows what to do here.

This is the repo: https://github.com/skibidibidop/progress

I tried to cancel the run but it fails every time
Github status says everything is okay
Attempts to push crash Neovim

**Comment by Blaz1kennBG** [comment]
+1. Im the same. I did see that github had some "red" stasuses and brushed it off but 10 hours after, github seems okay but the Action is still stuck haha

**Comment by yoannchaudet** [comment]
👋 I am following up with the Actions teams. This looks to be just a bad state following yesterday's outage. **UPDATE:** This is only a bad state in the system. It is not impacting other builds and deployments and will be cleared up automatically in the next 24 hours. You can just ignore this "queued" workflow for now.

A new commit on your repo should be triggering a new build properly. I did trigger on https://github.com/skibidibidop/progress/actions/runs/4325078564.

**Reply by skibidibidop** [reply]
Thank you very much!

**Reply by marcindulak** [reply]
Maybe in some cases queuing it's not due to an outage.

When a failed "pages-build-deployment" run is triggered manually using "Re-run jobs", it is queued instead of starting.
It may be a bug in "pages-build-deployment".

I experienced the following case.
First, fail a run of "pages-build-deployment", by disabling github actions, before the github pages are enabled.
Then enable github actions. Use "Re-run jobs" to trigger the run again manually, it is queued.
With the manually triggered run still queued, a new commit proceeds with an immediate "pages-build-deployment".

Example:
- manually triggered run, queued https://github.com/marcindulak/nba-teams-parallel-coordinates/actions/runs/12572158906
- commit triggered run, succeeded https://github.com/marcindulak/nba-teams-parallel-coordinates/actions/runs/12572240817

<img width="1383" alt="Screenshot 2025-01-01 at 17 11 55" src="https://github.com/user-attachments/assets/af072904-2b57-4836-b9c1-191c73e09a2f" />

**Reply by marcindulak** [reply]
Another report of queued state in https://github.com/orgs/community/discussions/140404 (maybe there are more).

**Reply by marcindulak** [reply]
For reference: The problematic run is still queued, after 2 months.

And to repeat the solution mentioned by others: with the manually triggered run still queued, a new commit proceeds with an immediate "pages-build-deployment".

![Screenshot from 2025-03-03 08-22-14](https://github.com/user-attachments/assets/ba715593-4ea5-4fcc-a149-dc22bc869da1)

**Reply by yashvesikar** [reply]
+1 this is still an issue for me as well with manually triggered runs still queued.

**Reply by sheikh-hashir14** [reply]
+2, Still an Issue here. First the Deployment Failed, on trying to re-run, It just Queued

**Reply by hahahotsoup** [reply]
well after 8 hours sleeping,i tried again and it deployed successfully,but guess what?its still a quened record in my blog's repository's action

**Reply by MrKeanu-V** [reply]
same, this is still an Issue for me, even though I manually triggered action run, but still queued

**Comment by JCG15** [comment]
Hola! a mi me pasa lo mismo, he intentado subir actualizaciones desde la aplicación del ordenador pero el workflow pages build and deployment se queda en Queued, he intentado subir una actualización hoy pero sigue igual y no se por qué puede ser, ayuda!!!

https://github.com/JCG15/JCG15.github.io/actions/runs/4555910424

**Reply by Lucoshi** [reply]
https://github.com/Lucoshi/Lucoshi-s-Video-Game-Palace/actions/runs/20275437285

**Comment by RanitManik** [comment]
![image](https://github.com/community/community/assets/138437760/8e2369ee-213f-4372-9700-1b3a7d3e4e31)

It's March 1st, 2024, and I am facing the same issue now. All of my GitHub pages are being queued, and I can't cancel any workflow runs.

**Reply by jonathanandrade10** [reply]
+1. same issue here now

**Comment by vadimkantorov** [comment]
Same issue:
<img width="708" alt="image" src="https://github.com/community/community/assets/1041752/1b8b6c4f-255a-4451-9abe-c0a62dbe9639">

**Comment by Kaartik7** [comment]
I am facing the same issue, I have triggered and re-triggered the deployment but it still shows queued up since a while. Any fix around that?
<img width="877" alt="Screenshot 2024-10-28 at 10 22 16 PM" src="https://github.com/user-attachments/assets/46fce54b-a579-495b-8139-7da1975cd417">

**Comment by IzzyYCCheng** [comment]
2024.10.30 and the same issue still happen to me
![image](https://github.com/user-attachments/assets/7c12911a-40fe-4310-9dac-611900a056ae)

**Reply by Futuza** [reply]
Seeing the same.  Looks like Github Actions being down is affecting page build/deployment as well, Github's status page is also not reporting the issue.  😢

**Reply by bray2sick** [reply]
> Seeing the same. Looks like Github Actions being down is affecting page build/deployment as well, Github's status page is also not reporting the issue. 😢

Here I am wondering why my pages won't deploy.

**Comment by ErnestEnki** [comment]
Same thing on 16/Dec/2024

![image](https://github.com/user-attachments/assets/38190b06-237b-4722-a9dc-e101c948cd29)

when I try to cancel the run, it says "Failed to cancel workflow."

I am not sure what else I can possibly do.

**Reply by rahulseetharaman** [reply]
Facing the same issue. Did you happen to resolve it @ErnestEnki ?

**Reply by ErnestEnki** [reply]
yes, after a lot of trial and error, I realized that there was no problem with my actions, what actually happened is that my account was shadow banned, that's why my actions wouldn't run.
I submitted a support ticket asking them to normalize my account, and they said it was because I had one of those alias emails that forwards to my real email, I did so for privacy and security concerns, but that gets you flagged. Once I changed my email to a more normal one, and submitted another ticket for them to verify, they reinstate my account and my actions again worked normally
not sure if it's the same for your case, but this was my case

**Reply by alexstanza** [reply]
I am facing the same issue here in 1 March 2025, how do I know if my account is shadow banned? and whether I should raise a support ticket?

**Comment by alexstanza** [comment]
I am facing the same issue here on 1 March 2025, I am confused whether I do anything wrong or I should just leave it there. A better guidance in a situation like this will improve the overall experience, since similar question has been posted since March 2023.

**Comment by pathak-aman** [comment]
Facing the same issue, 6th March 2025.
![image](https://github.com/user-attachments/assets/ab0e3d83-4b86-4f28-a7a7-27006d801b3f)

**Comment by wavky** [comment]
I re-ran a failed workflow, but it's still showing **queued** after 30 minutes. Other new deployments are also failing because of this workflow, and I can't cancel it.
<img width="2497" alt="Xnip2025-03-13_14-43-41" src="https://github.com/user-attachments/assets/0eda58ba-1634-4a2e-a9fe-91d1697536a9" />
<img width="1961" alt="Xnip2025-03-13_14-44-08" src="https://github.com/user-attachments/assets/e23600fd-08f7-46ba-a630-1f9c7ebf18e3" />

**Reply by wavky** [reply]
After I pushed a new commit, the new workflow deployment succeeded, but the old one is still queued. I’m not sure why, but the result looks fine.

**Comment by engineering87** [comment]
Facing the same issue, 22th April 2025.
Trying everything but always failed deploy.

```
Run actions/deploy-pages@v4
Fetching artifact metadata for "github-pages" in this workflow run
Found 1 artifact(s)
Creating Pages deployment with payload:
{
	"artifact_id": 2981923710,
	"pages_build_version": "62047c5e9479cff27795520a0d7d4313b9072807",
	"oidc_token": "***"
}
Error: Creating Pages deployment failed
Error: HttpError: Deployment request failed for 62047c5e9479cff27795520a0d7d4313b9072807 due to in progress deployment. Please cancel b54578b92eb832e96ae99ff842f01ce18bfd0fc5 first or wait for it to complete.
    at /home/runner/work/_actions/actions/deploy-pages/v4/node_modules/@octokit/request/dist-node/index.js:124:1
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at createPagesDeployment (/home/runner/work/_actions/actions/deploy-pages/v4/src/internal/api-client.js:125:1)
    at Deployment.create (/home/runner/work/_actions/actions/deploy-pages/v4/src/internal/deployment.js:74:1)
    at main (/home/runner/work/_actions/actions/deploy-pages/v4/src/index.js:30:1)
Error: Error: Failed to create deployment (status: 400) with build version 62047c5e9479cff27795520a0d7d4313b9072807. Request ID B881:1EBFBC:2AB0E6F:2C0E3D3:6806C54F Responded with: Deployment request failed for 62047c5e9479cff27795520a0d7d4313b9072807 due to in progress deployment. Please cancel b54578b92eb832e96ae99ff842f01ce18bfd0fc5 first or wait for it to complete.
```

**Comment by DrakelingTech** [comment]
Same problem on 15 May 2025. I can deploy just fine in all repositories but one.

**Comment by qualityshepherd** [comment]
Same for me just now... 

<img width="1029" alt="image" src="https://github.com/user-attachments/assets/1f389d13-6954-4a24-a392-eb6f05fa6e79" />

**Reply by qualityshepherd** [reply]
<img width="866" alt="image" src="https://github.com/user-attachments/assets/4517a686-1cc3-4bcc-8d5b-246340b23a82" />

**Reply by laurogripa** [reply]
Same here today.

**Comment by wowamin** [comment]
<img width="1284" height="311" alt="螢幕擷取畫面 2025-07-16 103936" src="https://github.com/user-attachments/assets/644d3720-44c5-4513-8012-b3a00a19cd99" />

**Reply by wowamin** [reply]
請問有人解決此問題了嗎?

**Reply by wowamin** [reply]
<img width="1517" height="615" alt="image" src="https://github.com/user-attachments/assets/d8f53eb8-c9af-4e42-973e-795e8d8128fb" />

**Reply by wowamin** [reply]
https://github.com/all-age/allage/actions/runs/16308848228

**Comment by pretentiousgit** [comment]
I fixed this by going to Settings and futzing with the Pages settings.
I had it set to deploy from the main branch and it wanted me to set up a yaml config with workflows to do something instead.
Github provides the yaml automatically when you select "Deploy using actions" and adds it to a little .github/static.yml file.

When I ran it that way, the thing immediately deployed; if nothing else, you can try it.

**Reply by bethac07** [reply]
In my case at my time, this didn't fix it - the failing action (`actions/deploy-pages@v4`) fails equally whether run as its own action or whether being run by the GH "deploy from a branch" auto-setup.

**Comment by Sieo** [comment]
<img width="1154" height="767" alt="image" src="https://github.com/user-attachments/assets/ce51b0e1-74d2-483b-ac9e-438265ac8dbc" />
Still the same error.
I have a jekyll-gh-pages.yml, I push onto the main branch, it build just fine, but can't deploy... I tried to cancel, queue, requeue, push, nothing seems to be working :/

Anyone got an idea ?

**Comment by mokurin000** [comment]
Seems Github pages deployment was broken again

**Comment by sentrychris** [comment]
This is becoming a recurring problem for me.

EDIT: This is getting worse. Every other deployment today looks like this:

<img width="310" height="583" alt="image" src="https://github.com/user-attachments/assets/7e54648d-39be-46f1-85b8-dcab89ae4744" />

---

EDIT 2: ...

<img width="563" height="143" alt="image" src="https://github.com/user-attachments/assets/6a754ccc-b163-44ad-9b53-afe7a0d32b69" />

**Reply by TryAgain911** [reply]
I have this exact same problem. Tried to run the actions from GitHub website, they are now stuck in queue status and it fails to cancel. Any help is welcomed!

**Reply by blu3fishez** [reply]
Same here. I'm facing that exact same issue as shown in the image.

**Reply by kallewesterling** [reply]
Same here -- last few days have been really bad :/

**Reply by epxzzy** [reply]
same here. it doesnt seem to resolve no matter what you do

**Reply by PiyalAhmed** [reply]
Seems like we are the fishes of the same pond. :')

**Reply by ctodd7977-rgb** [reply]
Why this problem can't be resolved

**Reply by anlesk** [reply]
same here!

**Reply by FeliipeFRA** [reply]
Same here, the problem came back?

**Comment by IzzyYCCheng** [comment]
[laugh]         Izzy YC Cheng reacted to your message:
________________________________

**Comment by tkfoss** [comment]
Still happening in dec 2025.

**Reply by Futuza** [reply]
Doesn't seem like Github is interested in fixing the issue or even investigating it.

**Comment by b-sik** [comment]
Still happening Jan 2026

4m 9s to deploy a single 41.9kb image file after redeploying a couple times.

`
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Current status: deployment_queued
Getting Pages deployment status...
Reported success!`

**Reply by al3xsus** [reply]
Yup, same problem. Site in Astro, build was successful, but deployment is stuck in a loop. Removed github-pages environment to force the runner to re-create everything from scratch, didn't help.

**Reply by uki-hub** [reply]
i faced the same issue, try to deploying to github pages

**Reply by tommhe14** [reply]
also seeing that my github pages run times out and fails, yet the status page says its functional;

https://www.githubstatus.com

**Reply by rvarunrathod** [reply]
Seeing same issue

**Reply by opcruz** [reply]
Same issue

**Reply by FeliipeFRA** [reply]
Same

**Reply by Pikachudy** [reply]
same

**Comment by omnixtar** [comment]
Latest: Waited a few hours. Rerun. Job completed.

Try this: https://omnixtar.github.io/moddadom/

====

Came here to report the same problem:

https://github.com/omnixtar/omnixtar.github.io/actions/runs/20967066462/job/60261614152

**Reply by justinbornais** [reply]
I am facing the same problem, there must be an outage of some kind.

**Comment by bethac07** [comment]
Had the same issue yesterday and today, yesterday just re-creating new content like 20 minutes later fixed the issue/got it onto a more stable runner, so far not so today. At least one today fully timed out after trying an failing deployment for 15 minutes.

**Comment by D1scoDancer** [comment]
Same problem exists today

**Reply by pfaj** [reply]
yup..

**Reply by elias-utf8** [reply]
Yes me too

**Reply by AlexAtkinson** [reply]
Oh Microsoft...
Their customers are so tolerant. Except Windows users -- it's Linux time. :/

**Reply by javiermorenof7** [reply]
Igual

**Reply by DEEPAKRNPGUPTA** [reply]
seems happening again ,,or only me

**Comment by AlexAtkinson** [comment]
This got it going for me in short order. I was using the classic publishing method.

1. Set: Settings > Pages > Source: [Github Actions]
2. Create: '.github/workflows/gh_pages_deploy.yaml'
```yaml
name: Deploy GitHub Pages

on:
  push:
    branches: ["main"]  # or your default branch
  workflow_dispatch:

permissions:
  contents: read
  pages: write
  id-token: write

concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Setup Pages
        uses: actions/configure-pages@v4

      # Add your build steps here if needed
      # For example, if using Jekyll:
      # - name: Build with Jekyll
      #   uses: actions/jekyll-build-pages@v1

      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          path: '.'  # or your build output directory

  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment. outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
```
3. Push

**Reply by tommhe14** [reply]
legend, thank you!

**Reply by justinbornais** [reply]
For me it was the opposite, I had it setup this way and it stopped deploying. I switched to Deploy from a Branch and after about an hour or so, it started working.

**Reply by AlexAtkinson** [reply]
Le sigh... This "fix" just happened to land on a momentarily clear queue for their deployment tooling.
................................ I am now grumpy.

**Reply by bethac07** [reply]
<s>The one advantage of having the action be explicit is that you can set a `with: timeout: ` to make it try for longer, but under the hood it's still the [same action](https://github.com/actions/deploy-pages) and for me I found no difference switching back and forth between the two while the issue was goign on (for me, it seems better now, but stochastic issues are always the hardest to debug!). So I went from "classic auto-deploy" -> "explicit action" -> back to the "classic" experience, but left the explicit one on a `workflow_dispatch`-only trigger so that if the normal barfs, I can just set the manual one to keep trying for a longer time period.</s> NVM, max-max timeout is only 10 minutes.

**Reply by AlexAtkinson** [reply]
It appears that once you're stuck in queue, you're done. Rerunning failed job's doesn't seem to fare any better.
Cancel the job. Do the following until it works.
  - `git commit --allow-empty -m "Damn you to Hell, Github!" && git push`

**Reply by bethac07** [reply]
<s>@AlexAtkinson Once you get an explicit fail, in my experience it refuses to try again (at least the "classic experience"), but it can and does sometimes succeed after 5 or 6 or 8 (or more) minutes if the action has never actually yet _failed_, so that's why explicitly giving a long timeout helped in my hands.</s> NVM, max-max timeout is only 10 minutes.

**Reply by AlexAtkinson** [reply]
It worked twice in a row, but now I'm stuck again in a queue. It's fine. I have to go put a roast in the smoker, anyhow.

**Reply by justinbornais** [reply]
> The one advantage of having the action be explicit is that you can set a `with: timeout: ` to make it try for longer, but under the hood it's still the [same action](https://github.com/actions/deploy-pages)

Doesn't the `actions/deploy-pages` have a maximum timeout of 10 minutes? From my experience, I was never able to increase the timeout.

**Reply by bethac07** [reply]
... yup, I missed a warning because I was doing "panic change all the things". NVM and ty for the double-check!
<img width="871" height="160" alt="image" src="https://github.com/user-attachments/assets/8938128e-c9d2-4bdb-85af-ebac4248642c" />

**Reply by justinbornais** [reply]
If only the `deployment_queued` time didn't count towards the 10 minute max. It's not our fault we're stuck in the queue.

**Comment by hahahotsoup** [comment]
<img width="2960" height="1848" alt="Screenshot_20260807_072705_Edge" src="https://github.com/user-attachments/assets/d5ac66ff-c7d2-4772-9963-6ff92a5dc248" />
2026.8.7 it still happened

**Reply by hahahotsoup** [reply]
well after 8 hours sleeping,i tried again and it deployed successfully,but guess what?its still a quened record in my blog's repository's action

**Reply by MrKeanu-V** [reply]
Same Issue, I have retriggered the same action three times, all with queued. 🥲

**Comment by Pikachudy** [comment]
邮件已收到！

**Reply by v-coder247** [reply]
Hi @Pikachudy, thanks for participating!

Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting to ensure discussions are adhering to our [Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md).

**Comment by MrKeanu-V** [comment]
<img width="1848" height="652" alt="image" src="https://github.com/user-attachments/assets/3bffbc46-018b-4676-a9aa-60a3738c083c" />
Same Issue! I have retriggered the same issue three times, all with queued. Anyone know how to fix it?👀

**Reply by MrKeanu-V** [reply]
<img width="1804" height="328" alt="image" src="https://github.com/user-attachments/assets/1cb12fb2-0975-41c9-b18d-51a321d280eb" />
A new action was successfully triggered, but this failed action is always in queued.

**Comment by Pikachudy** [comment]
邮件已收到！
